api/cluster.go

```diff
@@ -86,8 +87,10 @@ func (s *Server) indexList(ctx *middleware.Context, req models.IndexList) {
 func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 	series, err := s.getTargetsLocal(request.Requests)
 	if err != nil {
+		// the only errors returned are from us catching panics, so we should treat them
+		// all as internalServerErrors
```

Comment:
ok that all sounds good. but i still find it weird that in the caller we rely on knowledge of the implementation of the callee (getTargetsLocal) when deciding how to treat returned errors.
This is why I think it should be up to getTargetsLocal to describe the "kind" of error it's returning. (an internal error, or in the future other kinds of errors)
We could achieve that in two ways:

1) a protocol (http) agnostic error type, that has a few categories (internal error vs bad request vs service unavailable), which get converted into the http domain once we're in http specific code (e.g. getTargetsLocal is from dataprocessor which is http agnostic but cluster.go is in the http domain)

2) we could also have dataprocessor return specific http errors (400 vs 500 vs etc). even though dataprocessor is not "in the http domain", this would let us keep the code a bit simpler, because the abstract error type would look pretty similar to the http error type.